pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1044,72 +1044,81 @@ protected void internalDeleteTopic(boolean authoritative, boolean deleteSchema)
+                                                    .getAdminClient().topics()
+                                                    .getSubscriptionsAsync(topicName.getPartition(i).toString());
+                                            subscriptionFutures.add(subscriptionsAsync
+                                                    .thenApply(subscriptions::addAll));
+                                        }
+                                        resumeAsyncResponse(asyncResponse, subscriptions, subscriptionFutures);
                                     }
-                                });
-                            }).thenAccept(__ -> resumeAsyncResponse(asyncResponse, subscriptions, subscriptionFutures));
-                        } else {
-                            for (int i = 0; i < partitionMetadata.partitions; i++) {
-                                CompletableFuture<List<String>> subscriptionsAsync = pulsar().getAdminClient().topics()
-                                        .getSubscriptionsAsync(topicName.getPartition(i).toString());
-                                subscriptionFutures.add(subscriptionsAsync.thenApply(subscriptions::addAll));
+                                } catch (Exception e) {
+                                    log.error("[{}] Failed to get list of subscriptions for {}",
+                                            clientAppId(), topicName, e);
+                                    asyncResponse.resume(e);
+                                }
+                            } else {
+                                internalGetSubscriptionsForNonPartitionedTopic(asyncResponse, authoritative);
                             }
-                            resumeAsyncResponse(asyncResponse, subscriptions, subscriptionFutures);
-                        }
-                    } catch (Exception e) {
-                        log.error("[{}] Failed to get list of subscriptions for {}", clientAppId(), topicName, e);
-                        asyncResponse.resume(e);
+                        }).exceptionally(ex -> {
+                            log.error("[{}] Failed to get subscriptions for topic {}", clientAppId(), topicName, ex);
+                            resumeAsyncResponseExceptionally(asyncResponse, ex);
+                            return null;
+                        });
                     }
-                } else {
-                    internalGetSubscriptionsForNonPartitionedTopic(asyncResponse, authoritative);
-                }
-            }).exceptionally(ex -> {
-                log.error("[{}] Failed to get subscriptions for topic {}", clientAppId(), topicName, ex);
-                resumeAsyncResponseExceptionally(asyncResponse, ex);
-                return null;
-            });
-        }
+                }).exceptionally(ex -> {
+                    Throwable cause = ex.getCause();
+                    log.error("[{}] Failed to get subscriptions for topic {}", clientAppId(), topicName, cause);
```

Comment:
``[{}] Failed to  validate the global namespace/topic ownership while get subscriptions for topic {}``